frontend/src/lib/components/shared/Dropdown/Selectbox.tsx

```diff
@@ -20,24 +20,31 @@ import { Select as UISelect, OnChangeParams, Option } from "baseui/select"
 import { logWarning } from "src/lib/util/log"
 import VirtualDropdown from "src/lib/components/shared/Dropdown/VirtualDropdown"
 import { hasMatch, score } from "fzy.js"
+import { withTheme } from "@emotion/react"
+import { EmotionTheme } from "src/lib/theme"
 import _ from "lodash"
-import { LabelVisibilityOptions } from "src/lib/util/utils"
+import { isNullOrUndefined, LabelVisibilityOptions } from "src/lib/util/utils"
 import { Placement } from "src/lib/components/shared/Tooltip"
 import TooltipIcon from "src/lib/components/shared/TooltipIcon"
 import {
   WidgetLabel,
   StyledWidgetLabelHelp,
 } from "src/lib/components/widgets/BaseWidget"
 
+const NO_OPTIONS_MSG = "No options to select."
+const CHOOSE_OPTION_MSG = "Choose an option"
+
 export interface Props {
   disabled: boolean
   width?: number
-  value: number
-  onChange: (value: number) => void
+  value?: number
+  onChange: (value?: number) => void
   options: any[]
   label?: string | null
   labelVisibility?: LabelVisibilityOptions
   help?: string
+  theme: EmotionTheme
```

Comment:
It will be injected using `import { withTheme } from "@emotion/react"`, later on we have `export default withTheme(Selectbox)`, so it will be always present